docs/apache-airflow/administration-and-deployment/priority-weight.rst

```diff
@@ -24,6 +24,11 @@ Priority Weights
 bumped to any integer. Moreover, each task has a true ``priority_weight`` that is calculated based on its
 ``weight_rule`` which defines the weighting method used for the effective total priority weight of the task.
 
+.. versionadded:: 2.9.0
+
+    Total priority weight should be in range between **-2,147,483,648** and **2,147,483,647**.
```

Comment:
Do we support negative value for weight?

Reply:
The value doesn’t have any semantic meanings, things are simply ordered by it in the scheduler. A negative value works as well as positive, or zero.

Reply:
Thinking about this though, I wonder if we should just change the database field to a float instead. We don’t really care about the precise value here, and a float can be ordered as well as an int.

Reply:
That's an interesting thought. might be indeed simple to implement - just migration -  and does not require **any** code changes

Reply:
Yeah this is also applicable. It still required to change a code: Models + Migration

Just one nit this change might invoke internally recreate Task Instance table (delete old records, create new records). In user perspective it might required some time on huge TI table.

Reply:
With current implementation it is hard to overflow even `int32` (limits for MySQL and Postgres) priority of task + sum priorities of upstream/downstream tasks.

But it might changed in case of https://github.com/apache/airflow/pull/36029, some custom user defined WeightRules could easily overflow any values in case of progressive and exponential progressive implementations.

Reply:
Anyway, I will try to this approach, and we could decide which one is better suits for current and potential future implementations

Reply:
Even if we recommended it doesn't mean that it does not impact someone who even uses this recommendation.

This one only changes validation in one place and someone who does not use priorities probably does not even notice about this change but it introduces hard limits which already exist, but rather that crash scheduler it changed values to suitable.

The other one is required to change type, find which type better suits, write migrations and change types from int to float.